src/app/app.ts

```diff
@@ -70,4 +75,36 @@ client.once('ready', () => {
 // Login to Discord with your client's token
 client.login(process.env.DISCORD_BOT_TOKEN);
 
+// Listen to db sync events
+async function dbSyncListener(): Promise<void> {
+const db: Db = await MongoDbUtils.connect('bountyboard');
+const collection = db.collection("bounties");
+const changeStreamOptions: ChangeStreamOptions = { fullDocument: "updateLookup" };
+// This could be any pipeline.
+const pipeline = [];
+
+// const changeStream = collection.watch(pipeline, changeStreamOptions);
+const changeStream = collection.watch();
+
+// set up a listener when change events are emitted
+changeStream.on("change", async next => {
+	// note: passes full document, and not updated fields
+	Log.debug("received a change to the collection: \t" + JSON.stringify(next));
+	let changeEvent = next as unknown as ChangeStreamEvent;
+
+	// Note: for insert operation, fullDocument is populated instead of updatedFields
+	if (! ("insert" === changeEvent.operationType) ) {
+		const upsertedBountyRecord: BountyCollection = await collection.findOne({
+			_id: new mongo.ObjectId(changeEvent.documentKey._id),
+		});
+		changeEvent.fullDocument = upsertedBountyRecord;
+	}
+
+	ClientSync({ changeStreamEvent: changeEvent });
+	});
```

Comment:
To be clear here: we fetch the document by _id passed in changeEvent, then bind the full document to the changeEvent object, then instantiate a ClientSync object with that changeEvent - what happens to the ClientSync Object now?